jsonschema-generator/src/main/java/com/github/victools/jsonschema/generator/MemberScope.java

```diff
@@ -321,14 +322,21 @@ public <A extends Annotation> A getContainerItemAnnotationConsideringFieldAndGet
         if (this.isFakeContainerItemScope()) {
             return this.getContainerItemAnnotationConsideringFieldAndGetter(annotationClass);
         }
-        if (this.getOverriddenType() != null
-                && this.getDeclaredType().getErasedType() == Optional.class
-                && this.getOverriddenType().getErasedType() == this.getDeclaredType().getTypeParameters().get(0).getErasedType()) {
+        if (this.getOverriddenType() != null && (isSupplier() || isOptional())) {
             return this.getContainerItemAnnotationConsideringFieldAndGetter(annotationClass);
         }
         return null;
     }
 
+    private boolean isSupplier() {
+        return this.member.getType().isInstanceOf(Supplier.class);
+    }
+
+    private boolean isOptional() {
+        return this.getDeclaredType().getErasedType() == Optional.class
+                && this.getOverriddenType().getErasedType() == this.getDeclaredType().getTypeParameters().get(0).getErasedType();
+    }
```

Comment:
**suggestion**: While I agree with adding the support for the `Supplier` by default and appreciate the improved readability by extracting these two checks into dedicated methods, I'm not convinced by the method names. The `MemberScope` is neither a `Supplier` nor an `Optional`. It is the member's type that can be either of these two.
The method name should reflect that more clearly, even if that means it becoming more verbose.

Additionally, I'd also include the check of the supplied type against the member's overridden one – same as for `Optional` – and not indiscriminately consider all `Supplier` subtypes.